src/entrypoints/is-array.d.ts

```diff
@@ -1,3 +1,3 @@
 interface ArrayConstructor {
-  isArray(arg: any): arg is unknown[];
+  isArray<T>(arg: true extends TSReset.IsAny<T> ? never : T): arg is true extends TSReset.IsAny<T> ? never : T extends ReadonlyArray<unknown> ? T : Array<T> extends T ? Array<T> : never;
 }
```

Comment:
Not sure I agree with the decision to disallow `any`, since narrowing a value from `any` to `unknown[]` seems like a valid thing to want to do.

Setting that aside, here's another way you could accomplish the same thing:

```suggestion
interface ArrayConstructor {
  isArray<T>(arg: NotAny<T>): arg is Extract<NotAny<T>, readonly unknown[]>;
}

type NotAny<T> = 0 extends 1 & T ? never : T
```